tools/swarm/src/main.rs

```diff
@@ -1,63 +1,218 @@
 #![allow(missing_docs)]
 
-mod cli;
-mod compose;
-mod ui;
-mod util;
+/// Docker Compose configuration generator for Iroha.
+#[allow(clippy::struct_excessive_bools)]
+#[derive(clap::Parser, Debug)]
+pub struct Args {
+    // TODO: optional peer configuration file / other ways to configure peers
+    /// Number of peer services in the configuration.
+    #[arg(long, short, value_name = "COUNT")]
+    peers: std::num::NonZeroU16,
+    /// UTF-8 seed for deterministic key-generation.
+    #[arg(long, short)]
+    seed: Option<String>,
+    /// Includes a healthcheck for every service in the configuration.
+    ///
+    /// Healthchecks use predefined settings.
+    ///
+    /// For more details on healthcheck configuration in Docker Compose files, see:
+    /// <https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck>
+    #[arg(long, short = 'H')]
+    healthcheck: bool,
+    /// Directory with Iroha configuration.
+    /// It will be mapped to a volume for each container.
+    ///
+    /// The directory should contain `genesis.json` and the executor.
+    #[arg(long, short, value_name = "DIR")]
+    config_dir: std::path::PathBuf,
+    /// Docker image used by the peer services.
+    ///
+    /// By default, the image is pulled from Docker Hub if not cached.
+    /// Pass the `--build` option to build the image from a Dockerfile instead.
+    ///
+    /// **Note**: Swarm only guarantees that the Docker Compose configuration it generates
+    /// is compatible with the same Git revision it is built from itself. Therefore, if the
+    /// specified image is not compatible with the version of Swarm you are running,
+    /// the generated configuration might not work.
+    #[arg(long, short, value_name = "NAME")]
+    image: String,
+    /// Build the image from the Dockerfile in the specified directory.
+    /// Do not rebuild if the image has been cached.
+    ///
+    /// The provided path is resolved relative to the current working directory.
+    #[arg(long, short, value_name = "DIR")]
+    build: Option<std::path::PathBuf>,
+    /// Always pull or rebuild the image even if it is cached locally.
+    #[arg(long, short = 'C')]
+    no_cache: bool,
+    /// Path to the target Compose configuration file.
+    ///
+    /// If the file exists, the app will prompt its overwriting. If the TTY is not
+    /// interactive, the app will stop execution with a non-zero exit code.
+    /// To overwrite the file anyway, pass the `--force` flag.
+    #[arg(long, short, value_name = "FILE")]
+    out_file: std::path::PathBuf,
```

Comment:
If swarm supports both file output and STDOUT, I think it would be more CLI-idiomatic to use `--output` for file output and `-` for STDOUT. Not sure how to achieve it with clap.